### PR TITLE
refactor(vx-scan): Load machine config on backend

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview
Task: #2815 

Instead of loading the machine config from env variables in the proxy, load it in the backend and send it to the frontend via the API like any other backend data. This allows us to remove the last of the frontend app state.

_Review by commits_

## Demo Video or Screenshot
N/A

## Testing Plan 
- Updated automated tests

## Checklist
- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.
- [ ] I have added JSDoc comments to any newly introduced exports

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: apps/vx-scan/frontend/test/helpers/mock_api_client.ts [hunk 1]
```diff
@@ -10,10 +10,16 @@ import { createMockClient } from '@votingworks/grout-test-utils';
...
 import type {
   Api,
+  MachineConfig,
   PrecinctScannerConfig,
   PrecinctScannerStatus,
 } from '@votingworks/vx-scan-backend';
 
+export const machineConfig: MachineConfig = {
+  machineId: '0002',
+  codeVersion: '3.14',
```
Comment (arsalansufi) [inline apps/vx-scan/frontend/test/helpers/mock_api_client.ts:20]: 🥧

File: apps/vx-scan/backend/src/app.ts [hunk 2]
```diff
@@ -225,10 +220,14 @@ function buildApi(
...
-      machineId: string;
-    }): Promise<Result<void, ExportDataError>> {
-      return await exportCastVoteRecordsToUsbDrive(store, usb, input.machineId);
+    async exportCastVoteRecordsToUsbDrive(): Promise<
+      Result<void, ExportDataError>
+    > {
+      return await exportCastVoteRecordsToUsbDrive(
+        store,
+        usb,
+        getMachineConfig().machineId
```
Comment (arsalansufi) [inline apps/vx-scan/backend/src/app.ts:229]: Nice - loving that client calls are getting simpler as we consolidate data on the backend

File: apps/vx-scan/frontend/package.json [hunk 3]
```diff
@@ -79,10 +79,8 @@
     "dotenv": "8.2.0",
     "dotenv-expand": "5.1.0",
     "events": "^3.3.0",
-    "fetch-mock": "^9.11.0",
```
Comment (arsalansufi) [inline apps/vx-scan/frontend/package.json:82]: Wow!

**Review by arsalansufi — APPROVED** [review]
LGTM!

File: apps/vx-scan/frontend/package.json (hunk 3 above)
Comment (jonahkagan) [reply apps/vx-scan/frontend/package.json:81]: I knowwww. Dreams come true

**Review by eventualbuddha — APPROVED** [review]

**Pull request merged**